Sources/App/Core/Search.swift

```diff
@@ -255,14 +255,24 @@ enum Search {
 
         // only include non-package results on first page
         let query = page == 1
-        ? db.unionAll(
-            authorMatchQueryBuilder(on: database, terms: sanitizedTerms),
-            keywordMatchQueryBuilder(on: database, terms: sanitizedTerms),
-            packageMatchQueryBuilder(on: database, terms: sanitizedTerms, filters: filters,
-                                     offset: offset, limit: limit)
-        ).query
-        : packageMatchQueryBuilder(on: database, terms: sanitizedTerms, filters: filters,
-                                   offset: offset, limit: limit).query
+        ? authorMatchQueryBuilder(on: database, terms: sanitizedTerms)
+            .union(all: { _ in
+                keywordMatchQueryBuilder(on: database, terms: sanitizedTerms)
+
+            })
+            .union(all: { _ in
+                packageMatchQueryBuilder(on: database,
+                                         terms: sanitizedTerms,
+                                         filters: filters,
+                                         offset: offset,
+                                         limit: limit)
+
+            }).query
+        : packageMatchQueryBuilder(on: database,
+                                   terms: sanitizedTerms,
+                                   filters: filters,
+                                   offset: offset,
+                                   limit: limit).query
```

Comment:
I'm not really happy how this reads now but that's what the API reads like after review. It _is_ more expressive as it supports `... UNION ... UNION ALL ...`, which `db.unionAll(...)` doesn't but it _is_ making this a bit more awkward.

Reply:
Yea I agree this is a little harder to read.